### ISSUE TITLE
`Final` should not be compatible with `type[T]`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7094]
The following should generate an error but doesn't.

```python
T = TypeVar("T")

def takes_type(t: type[T]) -> T:
    return t()

takes_type(Final[int])
```

`Final` is a special form.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.349, which I just published.
